src/status/wallet.nim

```diff
@@ -50,6 +71,36 @@ proc initEvents*(self: WalletModel) =
 proc delete*(self: WalletModel) =
   discard
 
+proc trackPendingTransaction*(self: WalletModel, address: string, trxHash: string, trxType: PendingTransactionType, data: string) =
+  let latestBlock = getBlockByNumber("latest").parseJson()["result"].getInt
+
+  if not self.pendingTransactions.hasKey(address):
+    self.pendingTransactions[address] = @[]
+
+  self.pendingTransactions[address].add PendingTransaction(
+    transactionHash: trxHash,
+    trxType: trxType,
+    blockNumber: latestBlock,
+    data: data,
+    mined: false
+  )
+
+proc getTransactionReceipt*(self: WalletModel, transactionHash: string): JsonNode =
+  result = status_wallet.getTransactionReceipt(transactionHash).parseJSON()["result"]
+
+proc checkPendingTransactions*(self: WalletModel, address: string, blockNumber: int) =
+  if not self.pendingTransactions.hasKey(address): return
+  for trx in self.pendingTransactions[address].mitems:
+    if trx.mined: continue
+
+    let transactionReceipt = self.getTransactionReceipt(trx.transactionHash)
+    if transactionReceipt.kind != JNull:
+      trx.mined = true
+      if transactionReceipt{"status"}.getStr == "0x1": # mined successfully
+        self.events.emit(trx.trxType.event, TransactionMinedArgs(data: trx.data))
+      else:
+        discard # TODO: what should we do if the transaction reverted?
```

Comment:
Maybe we could change the `TransactionMinedArgs` to track the result as well, and then emit the event even on failure, so the UI can react accordingly:
```
type TransactionMinedArgs* = ref object of Args
  data*: string
  success*: bool
  revertReason*: string # possible to get revert reason in here?
```

Then we handle the event, we can check `success` and if successful, install the sticker pack, or if not successful, show an error message (hopefully with a revert reason).